src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/UniqueAccessCodeService.java

```diff
@@ -11,7 +12,17 @@
    *
    * @param uacHash hashed unique access code for which to retrieve data
    * @return UniqueAccessCodeDTO representing data for UAC
-   * @throws CTPException something wernt wrong
+   * @throws CTPException something went wrong
    */
   UniqueAccessCodeDTO getAndAuthenticateUAC(String uacHash) throws CTPException;
+
+  /**
+   * Link a UAC to a case and return a representation of the UAC.
+   *
+   * @param uacHash hashed unique access code for which to retrieve data.
+   * @param request contains the address detail to link to.
+   * @return UniqueAccessCodeDTO representing data for UAC.
+   * @throws CTPException something went wrong.
+   */
+  UniqueAccessCodeDTO linkUACCase(String uacHash, UACLinkRequestDTO request) throws CTPException;
```

Comment:
Here - this existing method needs to reverse it's mapping order as discussed Case FIRST and then UAC

Reply:
Setting logic reversed, by calling new createUniqueAccessCodeDTO() method